File: fizzbuzz.go
package main

import (
	"fmt"
	"strconv"
)

// If `num` is dividable by 3 and 5, return "FizzBuzz", if it is dividable by 3,
// return "Fizz", if it is dividable by 5, return "Buzz", for all the other
// cases, return the number.
func FizzBuzz(number int) string {
	switch {
	case (number%3 == 0) && (number%5 == 0):
		return "FizzBuzz"
	case number%3 == 0:
		return "Fizz"
	case number%5 == 0:
		return "Buzz"
	}
	return strconv.FormatInt(int64(number), 10)
}

// Iterate numbers from 1 to 100 and write result of `fizzBuzz` function to
// stdout.
func main() {
	for i := 1; i <= 100; i++ {
		fmt.Println(FizzBuzz(i))
	}
}

File: fizzbuzz.rb
def fizz_buzz(num)
  if num % 3 == 0 && num % 5 == 0
    puts "FizzBuzz"
  elsif num % 3 == 0
    puts "Fizz"
  elsif num % 5 == 0
    puts "Buzz"
  else
    puts "#{num}"
  end
end


def main
  (1..100).each do |number|
    print fizz_buzz(number)
  end
end


if __FILE__ == $0
  main
end
